services/efs-guardian/src/simcore_service_efs_guardian/services/background_tasks.py

```diff
@@ -1,18 +1,58 @@
 import logging
+from datetime import datetime, timezone
 
 from fastapi import FastAPI
+from models_library.projects import ProjectID
+from models_library.projects_state import ProjectStatus
+from servicelib.project_lock import (
+    PROJECT_LOCK_TIMEOUT,
+    PROJECT_REDIS_LOCK_KEY,
+    lock_project,
+)
+from simcore_postgres_database.utils_projects import ProjectsRepo
 
 from ..core.settings import ApplicationSettings
+from .efs_manager import EfsManager
+from .modules.redis import get_redis_lock_client
 
 _logger = logging.getLogger(__name__)
 
 
 async def removal_policy_task(app: FastAPI) -> None:
-    _logger.info("FAKE Removal policy task started (not yet implemented)")
-
-    # After X days of inactivity remove data from EFS
-    # Probably use `last_modified_data` in the project DB table
-    # Maybe lock project during this time lock_project()
+    _logger.info("Removal policy task started")
 
     app_settings: ApplicationSettings = app.state.settings
     assert app_settings  # nosec
+    efs_manager: EfsManager = app.state.efs_manager
+
+    base_start_timestamp = datetime.now(tz=timezone.utc)
+
+    efs_project_ids: list[
+        ProjectID
+    ] = await efs_manager.list_projects_across_whole_efs()
+
+    projects_repo = ProjectsRepo(app.state.engine)
+    for project_id in efs_project_ids:
+        _project_last_change_date = await projects_repo.get_project_last_change_date(
+            project_id
+        )
+        if (
+            _project_last_change_date is None
+            or _project_last_change_date
+            < base_start_timestamp
+            - app_settings.EFS_REMOVAL_POLICY_TASK_AGE_LIMIT_TIMEDELTA
+        ):
+            _logger.info(
```

Comment:
MINOR: using a context logger should provide you with info on start and end of this operaiton

Reply:
improved :+1: